app/models/accounting/quickbooks/fetcher.rb

```diff
@@ -0,0 +1,53 @@
+module Accounting
+  module Quickbooks
+    class Fetcher
+      def initialize(relation)
+        @relation = relation
+      end
+
+      def query
+        util = ::Quickbooks::Util::QueryBuilder.new
+
+        types.map do |type|
+          q = "select * from #{type} where #{util.clause('Id', 'in', ids)}"
+          service(type).query(q).entries
+        end.flatten
+      end
+
+      def populate(qb_objects)
+        @relation.each do |transaction|
+          qb = qb_objects.find { |qbo| qbo.id.to_s == transaction.qb_transaction_id }
+          transaction.qb_object = qb
+        end
+        @relation
+      end
+
+      def fetch
+        qb_objects = query
+        populate(qb_objects)
+      end
+
+      private
+
+      def ids
+        @relation.pluck(:qb_transaction_id)
+      end
+
+      def types
+        @relation.distinct(:qb_transaction_type).pluck(:qb_transaction_type)
+      end
+
+      def service(type)
+        ::Quickbooks::Service.const_get(type).new(auth_details)
+      end
+
+      def auth_details
+        { access_token: qb_connection.access_token, company_id: qb_connection.realm_id }
+      end
+
+      def qb_connection
+        @qb_connection ||= Division.root.qb_connection
+      end
+    end
+  end
+end
```

Comment:
It appears this class has no test coverage. This is concerning. We need to have good coverage on this core stuff.

What I recommend:

* We only need to test `fetch`
* Create a few transactions with two different types (using a factory)
* stub the `service` method to return a test double
* stub the `query` method on the `service` double and expect it to be called with the appropriate query methods (should be two, right?), and return a double with `entries` set to some qb object doubles with the appropriate `id`s set
* Instantiate with Transaction.all and expect that `fetch` returns the objects with the appropriate qb object doubles attached